packages/flutter_test/lib/src/finders.dart

```diff
@@ -302,19 +311,19 @@ class CommonFinders {
   /// ```dart
   /// expect(find.byWidgetPredicate(
   ///   (Widget widget) => widget is Tooltip && widget.message == 'Back',
-  ///   description: 'widget with tooltip "Back"',
+  ///   requirements: 'with tooltip "Back"',
   /// ), findsOneWidget);
   /// ```
   ///
-  /// If [description] is provided, then this uses it as the description of the
+  /// If [requirements] is provided, then this uses it as the description of the
   /// [Finder] and appears, for example, in the error message when the finder
   /// fails to locate the desired widget. Otherwise, the description prints the
   /// signature of the predicate function.
   ///
   /// If the `skipOffstage` argument is true (the default), then this skips
   /// nodes that are [Offstage] or that are from inactive [Route]s.
-  Finder byWidgetPredicate(WidgetPredicate predicate, { String? description, bool skipOffstage = true }) {
-    return _WidgetPredicateFinder(predicate, description: description, skipOffstage: skipOffstage);
+  Finder byWidgetPredicate(WidgetPredicate predicate, { String? description, String? requirements, bool skipOffstage = true }) {
```

Comment:
Should description be deprecated then? Also, not totally clear why this needs to be renamed? It is after all just a description of the predicate?

Reply:
description makes it clearer to me that this is a label for debugging only.

Reply:
Undid the various `description` -> `requirements` changes. Eventually I'd like to move the `description` parameters to match the `describeCandidate` parameters used by `CommonSemanticsFinders`, but for now the `description` stuff is good enough.